sos/report/plugins/dellrac.py

```diff
@@ -0,0 +1,63 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+import re
+import time
+
+
+class DellRAC(Plugin, IndependentPlugin):
+
+    short_desc = 'Dell Remote Access Controller Administration'
+
+    plugin_name = 'dellrac'
+    profiles = ('system', 'storage', 'hardware',)
+    packages = ('srvadmin-idracadm7',)
+
+    option_list = [
+        PluginOpt('debug', default=False, desc='capture support assist data')
+    ]
+
+    racadm = '/opt/dell/srvadmin/bin/idracadm7'
+
+    def setup(self):
+        for subcmd in ['getniccfg', 'getsysinfo']:
+            self.add_cmd_output(
+                '%s %s' % (self.racadm, subcmd),
+                suggest_filename='racadm_%s' % subcmd)
+
+        if self.get_option("debug"):
+            self.do_debug()
+
+    def do_debug(self):
+        self.add_cmd_output(
+            '%s supportassist accepteula',
+            suggest_filename='racadm_supportassist_accepteula')
```

Comment:
Does this actually accept the eula or does it only report on if it's been accepted?

If it's the former - absolutely no way.

Reply:
It does accept EULA if not already done. The `supportassist` sub command does not work if not already accepted.

Reply:
100% not going to happen in sos. We go through great lengths to not change system state, but we _absolutely_ cannot accept legal agreements on the behalf of users, not even if this were gated behind a plugin option.

Reply:
Sure. There is a eula status command. We can use that prior to collecting report. Okay?

Reply:
100% agree with @TurboTurtle - if users have to accept license terms in order to use functionality then that is something that we cannot automate. It's fine to check the status and log a warning message if that's possible, but we cannot perform an action that would agree to license terms on a user's behalf.

Reply:
Yup